sdk/src/main/java/ly/count/android/sdk/ModuleAPM.java

```diff
@@ -520,10 +519,19 @@ public void startTrace(String traceKey) {
          *
          * @param traceKey key by which this action is identified
          */
-        public void endTrace(String traceKey, Map<String, Integer> customMetrics) {
+        public void endTrace(@Nullable String traceKey, @Nullable Map<String, Integer> customMetrics) {
             synchronized (_cly) {
                 L.i("[Apm] Calling 'endTrace' with key:[" + traceKey + "]");
 
+                if (traceKey == null || traceKey.isEmpty()) {
+                    L.e("[Apm] Calling 'endTrace' with invalid traceKey");
+                    return;
+                }
+
+                if (customMetrics == null) {
+                    customMetrics = new ConcurrentHashMap<>();
```

Comment:
we can fix the codacy issue for this line

Reply:
We can fix this, but this will not differ really. I omitted this. But let me fix it then

Reply:
I feel like that codacy rule is not a beneficial codacy rule maybe we can omit it